src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,115 @@
+import codeu.model.store.basic.UserStore;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class AdminServletTest {
+
+  private HttpServletRequest mockRequest;
+  private HttpSession mockSession;
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+  private UserStore mockUserStore;
+  private ConversationStore mockConversationStore;
+  private MessageStore mockMessageStore;
+  private AdminServlet AdminServlet;
+
+    @Before
+    public void setup() {
+      AdminServlet = new AdminServlet();
+
+      mockRequest = Mockito.mock(HttpServletRequest.class);
+      mockSession = Mockito.mock(HttpSession.class);
+      Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+      mockResponse = Mockito.mock(HttpServletResponse.class);
+
+      mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+      Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/admin.jsp"))
+          .thenReturn(mockRequestDispatcher);
+
+      mockUserStore = Mockito.mock(UserStore.class);
+      AdminServlet.setUserStore(mockUserStore);
+
+      mockConversationStore = Mockito.mock(ConversationStore.class);
+      AdminServlet.setConversationStore(mockConversationStore);
+
+      mockMessageStore = Mockito.mock(MessageStore.class);
+      AdminServlet.setMessageStore(mockMessageStore);
+    }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+
+    List<User> fakeUserList = new ArrayList<>();
+    User fakeUser = new User(UUID.randomUUID(), "test_username", "fakePasswordHash", Instant.now());
+    fakeUser.setAdmin(true);
+    fakeUserList.add(fakeUser);
+
+    Mockito.when(mockUserStore.getAllUsers()).thenReturn(fakeUserList);
+
+    UUID fakeConversationId = UUID.randomUUID();
+    List<Conversation> fakeConversationList = new ArrayList<>();
+    fakeConversationList.add(
+        new Conversation(UUID.randomUUID(), UUID.randomUUID(), "test_conversation", Instant.now()));
+    Mockito.when(mockConversationStore.getAllConversations()).thenReturn(fakeConversationList);
+
+    List<Message> fakeMessageList = new ArrayList<>();
+    fakeMessageList.add(
+        new Message(UUID.randomUUID(),fakeConversationId, UUID.randomUUID(),"test_message",Instant.now()));
+    Mockito.when(mockMessageStore.getAllMessages()).thenReturn(fakeMessageList);
+    AdminServlet.doGet(mockRequest, mockResponse);
+
+    //Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
```

Comment:
Maybe verify if it set those attributes and goes to admin.jsp.

Reply:
Also, I'm not sure but in your doGet() doesn't it call getAttribute("user") on request object. How come you don't call when() for that and return the fake user you made to maybe check if it correctly identifies admins?